warehouse/balancers/src/main/java/datawave/ingest/table/balancer/ShardedTableTabletBalancer.java

```diff
@@ -93,23 +93,21 @@ protected Iterable<Pair<KeyExtent,Location>> getLocationProvider() {
     @Override
     protected int getMaxMigrations() {
         int maxMigrations = MAX_MIGRATIONS_DEFAULT;
+        String maxMigrationsProp = this.configuration.getTableConfiguration(this.tableId).get(SHARDED_MAX_MIGRATIONS);
         try {
-            String maxMigrationsProp = this.configuration.getTableConfiguration(this.tableId).get(SHARDED_MAX_MIGRATIONS);
             if (maxMigrationsProp != null && !maxMigrationsProp.isEmpty()) {
-                try {
-                    maxMigrations = Integer.parseInt(maxMigrationsProp);
-                } catch (Exception e) {
-                    log.error("Unable to parse " + SHARDED_MAX_MIGRATIONS + " value (" + maxMigrationsProp + ") as an integer.  Defaulting to " + maxMigrations);
-                }
+                maxMigrations = Integer.parseInt(maxMigrationsProp);
             }
+        } catch (NumberFormatException e) {
+            log.error("Unable to parse " + SHARDED_MAX_MIGRATIONS + " value (" + maxMigrationsProp + ") as an integer.  Defaulting to " + maxMigrations);
         } catch (Exception e) {
             log.warn("Failed to get " + SHARDED_MAX_MIGRATIONS + ".  Defaulting to " + maxMigrations, e);
         }
```

Comment:
This isn't exactly equivalent to what we had.  Consider doing something like this:
```suggestion
        } catch (Exception e) {
            if (e instanceof NumberFormatException) {
                log.error("Unable to parse " + SHARDED_MAX_MIGRATIONS + " value (" + maxMigrationsProp + ") as an integer.  Defaulting to " + maxMigrations);
            }
            log.warn("Failed to get " + SHARDED_MAX_MIGRATIONS + ".  Defaulting to " + maxMigrations, e);
        }
```

Reply:
Made suggested change.